integration-tests/tests/src/tests/mixed.ts

```diff
@@ -0,0 +1,313 @@
+    });
+
+    it("support complex primitive types", function (this: RealmContext) {
+      const d128 = Decimal128.fromString("6.022e23");
+      const oid = new ObjectId();
+      const uuid = new UUID();
+      const date = new Date();
+
+      const data: ISingleSchema = this.realm.write(() =>
+        this.realm.create(SingleSchema.name, { a: oid, b: uuid, c: d128, d: date }),
+      );
+
+      expect(typeof data.a === typeof oid, "should be the same type ObjectId");
+      expect(String(data.a)).equals(oid.toString(), "should be the same ObjectId");
+      expect(String(data.b)).equals(uuid.toString(), "Should be the same UUID");
+      expect(String(data.c)).equals(d128.toString(), "Should be the same Decimal128");
+      expect(String(data.d)).equals(date.toString(), "Should be the same Date");
+    });
+
+    it("support mixed mutability", function (this: RealmContext) {
+      const d128 = Decimal128.fromString("6.022e23");
+      const oid = new ObjectId();
+      const uuid = new UUID();
+      const date = new Date();
+
+      const data: ISingleSchema = this.realm.write(() =>
+        this.realm.create<ISingleSchema>(SingleSchema.name, { a: oid }),
+      );
+      expect(typeof data.a === typeof oid, "should be the same type ObjectId");
+      expect(String(data.a)).equals(oid.toString(), "should have the same content");
+
+      this.realm.write(() => (data.a = uuid));
+      expect(typeof data.a === typeof uuid, "should be the same type UUID");
+      expect(String(data.a)).equals(uuid.toString(), "should have the same content UUID");
+
+      this.realm.write(() => (data.a = d128));
+      expect(String(data.a)).equals(d128.toString(), "Should be the same Decimal128");
+
+      this.realm.write(() => (data.a = 12345678));
+      expect(data.a).equals(12345678, "Should be the same 12345678");
+
+      this.realm.write(() => ((data.a = null), "Should be the same null"));
+      expect(data.a).equals(null);
+
+      this.realm.write(() => ((data.a = undefined), "Should be the same null"));
+      expect(data.a).equals(null);
+    });
+    it("wrong type throws", function (this: RealmContext) {
+      expect(() => {
+        this.realm.write(() => this.realm.create(SingleSchema.name, { a: Object.create({}) }));
+      }).throws(Error, "Only Realm instances are supported.");
+    });
+  });
+
+  describe("Nested types", () => {
+    openRealmBefore({ schema: [SingleSchema, VertexSchema, MixNestedSchema] });
+    it("support nested types", function (this: RealmContext) {
+      this.realm.write(() => {
+        const r = this.realm.create(VertexSchema.name, { a: 1, b: 0, c: 0 });
+        const r2 = this.realm.create(VertexSchema.name, { a: 0, b: 1, c: 0 });
+        const r3 = this.realm.create(VertexSchema.name, { a: 0, b: 0, c: 1 });
+
+        this.realm.create(SingleSchema.name, { a: r, b: r2, c: r3, d: null });
+      });
+
+      let data: ISingleSchema = this.realm.objects<ISingleSchema>(SingleSchema.name)[0];
```

Comment:
Doing it this way, your test would be more robust to other tests that might write to the collection before it.

```suggestion
      let data = this.realm.write(() => {
        const r = this.realm.create(VertexSchema.name, { a: 1, b: 0, c: 0 });
        const r2 = this.realm.create(VertexSchema.name, { a: 0, b: 1, c: 0 });
        const r3 = this.realm.create(VertexSchema.name, { a: 0, b: 0, c: 1 });

        this.realm.create<ISingleSchema>(SingleSchema.name, { a: r, b: r2, c: r3, d: null });
      });
```